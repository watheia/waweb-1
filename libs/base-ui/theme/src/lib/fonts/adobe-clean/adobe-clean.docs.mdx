---
description: 'Adobe Clean Fonts'
labels: ['spectrum', 'typography', 'webfonts']
---

import {
  AdobeCleanLight,
  AdobeCleanLightIt,
  AdobeCleanSemiLight,
  AdobeCleanSemiLightIt,
  AdobeClean,
  AdobeCleanIt,
  AdobeCleanBold,
  AdobeCleanBoldIt,
  AdobeCleanExtraBold,
  AdobeCleanExtraBoldIt,
  AdobeCleanBlack,
  AdobeCleanBlackIt,
  adobeClean,
} from '.';

Provides individual font faces for the `Adobe Clean` font family.

### Component usage

Font faces are loaded on demand. Each font class needs to be added somewhere
in the entry point in order to use a specific variant. Use the `Adobe Clean`
font family, or any of the exported class names to style text blocks

```jsx live
<div>
  <AdobeCleanLight />
  <AdobeCleanLightIt />
  <AdobeCleanSemiLight />
  <AdobeCleanSemiLightIt />
  <AdobeClean />
  <AdobeCleanIt />
  <AdobeCleanBold />
  <AdobeCleanBoldIt />
  <AdobeCleanExtraBold />
  <AdobeCleanExtraBoldIt />
  <AdobeCleanBlack />
  <AdobeCleanBlackIt />
  <div style={{ fontFamily: 'Adobe Clean' }}>
    <p>
      <strong>abcdefghijklmnopqrstuvwxyz</strong>
      <br />
      <em>ABCDEFGHIJKLMNOPQRSTUVWXYZ</em>
      <br />
      <span style={{ fontWeight: 300 }}>
        0123456789.:,;()*!?'@#&lt;&gt;$%&amp;^+-=~
      </span>
    </p>
    <p className={adobeClean.light}>
      The quick brown fox jumps over the lazy dog.
    </p>
    <p className={adobeClean.lightIt}>
      The quick brown fox jumps over the lazy dog.
    </p>
    <p className={adobeClean.semiLight}>
      The quick brown fox jumps over the lazy dog.
    </p>
    <p className={adobeClean.semiLightIt}>
      The quick brown fox jumps over the lazy dog.
    </p>
    <p className={adobeClean.it}>
      The quick brown fox jumps over the lazy dog.
    </p>
    <p className={adobeClean.bold}>
      The quick brown fox jumps over the lazy dog.
    </p>
    <p className={adobeClean.boldIt}>
      The quick brown fox jumps over the lazy dog.
    </p>
    <p className={adobeClean.extraBold}>
      The quick brown fox jumps over the lazy dog.
    </p>
    <p className={adobeClean.extraBoldIt}>
      The quick brown fox jumps over the lazy dog.
    </p>
    <p className={adobeClean.black}>
      The quick brown fox jumps over the lazy dog.
    </p>
    <p className={adobeClean.blackIt}>
      The quick brown fox jumps over the lazy dog.
    </p>
  </div>
</div>
```
