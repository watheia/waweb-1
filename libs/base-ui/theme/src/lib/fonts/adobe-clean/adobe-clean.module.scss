.light {
  font-family: 'Adobe Clean';
  font-weight: 300;
  font-style: normal;
}

.lightIt {
  font-family: 'Adobe Clean';
  font-weight: 300;
  font-style: italic;
}

.semiLightIt {
  font-family: 'Adobe Clean';
  font-weight: 300;
  font-style: italic;
}

.semiLight {
  font-family: 'Adobe Clean';
  font-weight: 300;
  font-style: normal;
}

.regular {
  font-family: 'Adobe Clean';
  font-weight: normal;
  font-style: normal;
}

.it {
  font-family: 'Adobe Clean';
  font-weight: normal;
  font-style: italic;
}

.medium {
  font-family: 'Adobe Clean';
  font-weight: 500;
  font-style: normal;
}

.mediumIt {
  font-family: 'Adobe Clean';
  font-weight: 500;
  font-style: italic;
}

.extraBoldIt {
  font-family: 'Adobe Clean';
  font-weight: bold;
  font-style: italic;
}

.extraBold {
  font-family: 'Adobe Clean';
  font-weight: bold;
  font-style: normal;
}

.extraBoldIt {
  font-family: 'Adobe Clean';
  font-weight: bold;
  font-style: italic;
}

// .semiCnIt {
//   font-family: 'Adobe Clean';
//   font-weight: normal;
//   font-style: normal;
// }

// .semiCn {
//   font-family: 'Adobe Clean';
//   font-weight: normal;
//   font-style: italic;
// }

// .cond {
//   font-family: 'Adobe Clean';
//   font-weight: normal;
//   font-style: normal;
// }

// .condIt {
//   font-family: 'Adobe Clean';
//   font-weight: normal;
//   font-style: italic;
// }

// .boldSemiCn {
//   font-family: 'Adobe Clean';
//   font-weight: bold;
//   font-style: normal;
// }

// .boldSemiCnIt {
//   font-family: 'Adobe Clean';
//   font-weight: bold;
//   font-style: italic;
// }

// .boldCond {
//   font-family: 'Adobe Clean';
//   font-weight: bold;
//   font-style: normal;
// }

// .boldCondIt {
//   font-family: 'Adobe Clean';
//   font-weight: bold;
//   font-style: italic;
// }

.boldIt {
  font-family: 'Adobe Clean';
  font-weight: bold;
  font-style: italic;
}

.bold {
  font-family: 'Adobe Clean';
  font-weight: bold;
  font-style: normal;
}

.black {
  font-family: 'Adobe Clean';
  font-weight: 900;
  font-style: normal;
}

.blackItalic {
  font-family: 'Adobe Clean';
  font-weight: 900;
  font-style: italic;
}
