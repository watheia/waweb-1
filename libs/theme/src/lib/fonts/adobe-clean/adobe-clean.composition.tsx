import React from 'react';
import {
  AdobeCleanBlack,
  AdobeCleanBlackIt,
  AdobeCleanBoldIt,
  AdobeCleanExtraBold,
  AdobeCleanExtraBoldIt,
  AdobeCleanIt,
  AdobeCleanLight,
  AdobeCleanLightIt,
  AdobeClean,
  AdobeCleanSemiLight,
  AdobeCleanSemiLightIt,
  AdobeCleanBold,
} from './adobe-clean';

export const MockComponent = () => (
  <div data-testid="waweb.spectrum/theme/fonts/adobe-clean">
    <p>
      <strong>abcdefghijklmnopqrstuvwxyz</strong>
      <br />
      <em>ABCDEFGHIJKLMNOPQRSTUVWXYZ</em>
      <br />
      <span style={{ fontWeight: 300 }}>
        0123456789.:,;()*!?'@#&lt;&gt;$%&amp;^+-=~
      </span>
    </p>
    <p style={{ fontSize: '10px' }}>
      The quick brown fox jumps over the lazy dog.
    </p>
    <p style={{ fontSize: '11px' }}>
      The quick brown fox jumps over the lazy dog.
    </p>
    <p style={{ fontSize: '12px' }}>
      The quick brown fox jumps over the lazy dog.
    </p>
    <p style={{ fontSize: '14px' }}>
      The quick brown fox jumps over the lazy dog.
    </p>
    <p style={{ fontSize: '18px' }}>
      The quick brown fox jumps over the lazy dog.
    </p>
    <p style={{ fontSize: '24px' }}>
      The quick brown fox jumps over the lazy dog.
    </p>
    <p style={{ fontSize: '30px' }}>
      The quick brown fox jumps over the lazy dog.
    </p>
    <p style={{ fontSize: '36px' }}>
      The quick brown fox jumps over the lazy dog.
    </p>
    <p style={{ fontSize: '48px' }}>
      The quick brown fox jumps over the lazy dog.
    </p>
    <p style={{ fontSize: '60px' }}>
      The quick brown fox jumps over the lazy dog.
    </p>
    <p style={{ fontSize: '72px' }}>
      The quick brown fox jumps over the lazy dog.
    </p>
  </div>
);

export const AdobeCleanLightComp = () => (
  <AdobeCleanLight>
    <MockComponent />
  </AdobeCleanLight>
);

export const AdobeCleanLightItComp = () => (
  <AdobeCleanLightIt>
    <MockComponent />
  </AdobeCleanLightIt>
);

export const AdobeCleanSemiLightComp = () => (
  <AdobeCleanSemiLight>
    <MockComponent />
  </AdobeCleanSemiLight>
);

export const AdobeCleanSemiLightItComp = () => (
  <AdobeCleanSemiLightIt>
    <MockComponent />
  </AdobeCleanSemiLightIt>
);

export const AdobeCleanComp = () => (
  <AdobeClean>
    <MockComponent />
  </AdobeClean>
);

export const AdobeCleanItComp = () => (
  <AdobeCleanIt>
    <MockComponent />
  </AdobeCleanIt>
);

export const AdobeCleanBoldComp = () => (
  <AdobeCleanBold>
    <MockComponent />
  </AdobeCleanBold>
);
export const AdobeCleanBoldItComp = () => (
  <AdobeCleanBoldIt>
    <MockComponent />
  </AdobeCleanBoldIt>
);

export const AdobeCleanExtraBoldComp = () => (
  <AdobeCleanExtraBold>
    <MockComponent />
  </AdobeCleanExtraBold>
);

export const AdobeCleanExtraBoldItComp = () => (
  <AdobeCleanExtraBoldIt>
    <MockComponent />
  </AdobeCleanExtraBoldIt>
);

export const AdobeCleanBlackComp = () => (
  <AdobeCleanBlack>
    <MockComponent />
  </AdobeCleanBlack>
);

export const AdobeCleanBlackItComp = () => (
  <AdobeCleanBlackIt>
    <MockComponent />
  </AdobeCleanBlackIt>
);
