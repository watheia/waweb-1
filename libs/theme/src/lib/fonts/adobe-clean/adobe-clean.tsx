import React, { ReactNode, HtmlHTMLAttributes } from 'react';
import { Global, css } from '@emotion/react';
import { camelCase } from 'lodash';
import styles from './adobe-clean.module.scss';

export const staticFontsPath =
  'https://storage.googleapis.com/watheia.io/assets/fonts';

export type FontKerning = 'normal' | 'semi-condensed' | 'condensed';

export type FontWeight =
  | 'light'
  | 'semi-light'
  | 'normal'
  | 'bold'
  | 'extra-bold'
  | 'black';

export type FontStyle = 'italic' | 'normal';

export type AdobeCleanClass =
  | 'light'
  | 'lightIt'
  | 'semiLightIt'
  | 'semiLight'
  | 'regular'
  | 'it'
  | 'bold'
  | 'boldIt'
  | 'extraBold'
  | 'extraBoldIt'
  | 'black'
  | 'blackIt';

export const fontWeights: Record<FontWeight, string | number> = {
  light: 300,
  'semi-light': 300,
  normal: 'normal',
  bold: 'bold',
  'extra-bold': 'bold',
  black: 900,
};

// TODO this can be done more programaticly
const fontName = (weight: FontWeight, style: FontStyle): string => {
  let name = '';
  if (weight === 'black' && style === 'italic') name = 'BlackIt';
  else if (weight === 'black' && style === 'normal') name = 'Black';
  else if (weight === 'extra-bold' && style === 'normal') name = 'ExtraBold';
  else if (weight === 'extra-bold' && style === 'italic') name = 'ExtraBoldIt';
  else if (weight === 'bold' && style === 'normal') name = 'Bold';
  else if (weight === 'bold' && style === 'italic') name = 'BoldIt';
  else if (weight === 'normal' && style === 'italic') name = 'It';
  else if (weight === 'normal' && style === 'normal') name = 'Regular';
  else if (weight === 'light' && style === 'italic') name = 'LightIt';
  else if (weight === 'light' && style === 'normal') name = 'Light';
  else if (weight === 'semi-light' && style === 'italic') name = 'SemiLightIt';
  else if (weight === 'semi-light' && style === 'normal') name = 'SemiLight';
  else throw new Error('Invalid font varient');

  return name;
};

export type AdobeCleanProps = {
  fontStyle?: FontStyle;
  weight?: FontWeight;
} & HtmlHTMLAttributes<HTMLDivElement>;

export const AdobeClean = ({
  weight = 'normal',
  fontStyle = 'normal',
  className,
  children,
  ...props
}: AdobeCleanProps) => {
  const font = fontName(weight, fontStyle);
  className = className ?? camelCase(font);
  return (
    <div className={styles[className]} {...props}>
      <Global
        styles={css`
          @font-face {
            font-family: 'Adobe Clean';
            src: url('${staticFontsPath}/subset-AdobeClean-${font}.woff2')
                format('woff2'),
              url('${staticFontsPath}/subset-AdobeClean-${font}.woff')
                format('woff');
            font-weight: ${fontWeights[weight]};
            font-style: ${fontStyle};
            font-display: swap;
          }
        `}
      />
      <link
        rel="preload"
        href={`${staticFontsPath}/subset-AdobeClean-${font}.woff2`}
        as="font"
        type="font/woff2"
      />
      {children}
    </div>
  );
};

export const AdobeCleanIt = ({ children }: AdobeCleanProps) => (
  <AdobeClean fontStyle="italic">{children}</AdobeClean>
);

export const AdobeCleanLight = ({ children }: AdobeCleanProps) => (
  <AdobeClean weight="light">{children}</AdobeClean>
);

export const AdobeCleanLightIt = ({ children }: AdobeCleanProps) => (
  <AdobeClean weight="light" fontStyle="italic">
    {children}
  </AdobeClean>
);

export const AdobeCleanSemiLight = ({ children }: AdobeCleanProps) => (
  <AdobeClean weight="semi-light">{children}</AdobeClean>
);

export const AdobeCleanSemiLightIt = ({ children }: AdobeCleanProps) => (
  <AdobeClean weight="semi-light" fontStyle="italic">
    {children}
  </AdobeClean>
);

export const AdobeCleanBold = ({ children }: AdobeCleanProps) => (
  <AdobeClean weight="bold">{children}</AdobeClean>
);

export const AdobeCleanBoldIt = ({ children }: AdobeCleanProps) => (
  <AdobeClean weight="bold" fontStyle="italic">
    {children}
  </AdobeClean>
);

export const AdobeCleanExtraBold = ({ children }: AdobeCleanProps) => (
  <AdobeClean weight="extra-bold">{children}</AdobeClean>
);

export const AdobeCleanExtraBoldIt = ({ children }: AdobeCleanProps) => (
  <AdobeClean weight="extra-bold" fontStyle="italic">
    {children}
  </AdobeClean>
);

export const AdobeCleanBlack = ({ children }: AdobeCleanProps) => (
  <AdobeClean weight="black">{children}</AdobeClean>
);

export const AdobeCleanBlackIt = ({ children }: AdobeCleanProps) => (
  <AdobeClean weight="black" fontStyle="italic">
    {children}
  </AdobeClean>
);
