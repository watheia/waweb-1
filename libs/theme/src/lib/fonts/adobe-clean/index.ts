import _styles from './adobe-clean.module.scss';

export type { AdobeCleanProps, FontWeight, FontStyle } from './adobe-clean';
export * from './adobe-clean';

const {
  light,
  lightIt,
  semiLightIt,
  semiLight,
  regular,
  it,
  bold,
  boldIt,
  extraBold,
  extraBoldIt,
  black,
  blackIt,
} = _styles;

export const adobeClean = {
  light,
  lightIt,
  semiLightIt,
  semiLight,
  regular,
  it,
  bold,
  boldIt,
  extraBold,
  extraBoldIt,
  black,
  blackIt,
};
