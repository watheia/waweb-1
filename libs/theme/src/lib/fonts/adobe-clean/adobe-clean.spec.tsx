import React from 'react';
import { render } from '@testing-library/react';
import { AdobeCleanComp } from './adobe-clean.composition';

it('should render with the correct font', () => {
  const { getByTestId } = render(<AdobeCleanComp />);
  const container = getByTestId('waweb.spectrum/theme/fonts/adobe-clean');
  expect(container).toBeTruthy();
  expect(container).toMatchSnapshot();
});
