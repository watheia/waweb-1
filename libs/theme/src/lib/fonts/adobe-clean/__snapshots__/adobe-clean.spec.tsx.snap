// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with the correct font 1`] = `
<div
  data-testid="waweb.spectrum/theme/fonts/adobe-clean"
>
  <p>
    <strong>
      abcdefghijklmnopqrstuvwxyz
    </strong>
    <br />
    <em>
      ABCDEFGHIJKLMNOPQRSTUVWXYZ
    </em>
    <br />
    <span
      style="font-weight: 300;"
    >
      0123456789.:,;()*!?'@#&lt;&gt;$%&^+-=~
    </span>
  </p>
  <p
    style="font-size: 10px;"
  >
    The quick brown fox jumps over the lazy dog.
  </p>
  <p
    style="font-size: 11px;"
  >
    The quick brown fox jumps over the lazy dog.
  </p>
  <p
    style="font-size: 12px;"
  >
    The quick brown fox jumps over the lazy dog.
  </p>
  <p
    style="font-size: 14px;"
  >
    The quick brown fox jumps over the lazy dog.
  </p>
  <p
    style="font-size: 18px;"
  >
    The quick brown fox jumps over the lazy dog.
  </p>
  <p
    style="font-size: 24px;"
  >
    The quick brown fox jumps over the lazy dog.
  </p>
  <p
    style="font-size: 30px;"
  >
    The quick brown fox jumps over the lazy dog.
  </p>
  <p
    style="font-size: 36px;"
  >
    The quick brown fox jumps over the lazy dog.
  </p>
  <p
    style="font-size: 48px;"
  >
    The quick brown fox jumps over the lazy dog.
  </p>
  <p
    style="font-size: 60px;"
  >
    The quick brown fox jumps over the lazy dog.
  </p>
  <p
    style="font-size: 72px;"
  >
    The quick brown fox jumps over the lazy dog.
  </p>
</div>
`;
